### ISSUE TITLE
what is the function of  'network.outputs = tf.nn.lrn(network.outputs, 4,     )'

### ISSUE THREAD

**Issue by luoruisichuan** [zsdonghao/tensorlayer#21]
hi, @zsdonghao 
in tensorlayer/example/tutorial_cifar10_tfrecord.py, there is 
''
network.outputs = tf.nn.lrn(network.outputs, 4, bias=1.0, alpha=0.001 / 9.0,
                                                            beta=0.75, name='norm1')
''
the input is 'network.outputs', but other layer's input is 'network', for example
''
network = tl.layers.PoolLayer(network,
                                ksize=[1, 3, 3, 1],
                                strides=[1, 2, 2, 1],
                                padding='SAME',
                                pool = tf.nn.max_pool,
                                name ='pool_layer2') 
''
my question is that, what is the function of above sentence(network.outputs), and what is the difference between the two kinds inputs?
thank you !

**Comment by zsdonghao** [comment]
The ``network.outputs`` is the output tensor of the current layer.

The ``tf.nn.lrn`` is the shortcut of ``tf.nn.local_response_normalization``, which means apply local response normalization to the output.

We provide ``LocalResponseNormLayer``, but as local response normalization does not have any weights and arguments, you can also apply ``tf.nn.lrn`` on ``network.outputs``.

You may find the following links will help:
[Understanding basic layer](http://tensorlayer.readthedocs.io/en/latest/modules/layers.html)
[Understanding dense layer](http://tensorlayer.readthedocs.io/en/latest/modules/layers.html#understand-dense-layer)
[How to learn TensorLayer](http://tensorlayer.readthedocs.io/en/latest/user/more.html#how-to-effectively-learn-tensorlayer)

**Comment by luoruisichuan** [comment]
what is the difference between the two kinds inputs? one is 'network' and the other is 'network.outputs' in  tensorlayer/example/tutorial_cifar10_tfrecord.py.

**Comment by zsdonghao** [comment]
@luoruisichuan `network` is the object of a TensorLayer's layer, `network.outputs` is the tensor output of TensorFlow.

Again, you may find the following links will help:
[Understanding basic layer](http://tensorlayer.readthedocs.io/en/latest/modules/layers.html)
[Understanding dense layer](http://tensorlayer.readthedocs.io/en/latest/modules/layers.html#understand-dense-layer)
[How to learn TensorLayer](http://tensorlayer.readthedocs.io/en/latest/user/more.html#how-to-effectively-learn-tensorlayer)

**Comment by luoruisichuan** [comment]
Thank you.
I see.
"network = tl.layers.InputLayer(x, name='input_layer')"  translate tensorlayer instance into tensorflow instance,
and
"network = tl.layers.InputLayer(x, name='input_layer')"  ranslate tensorflow instance into tensorlayer instance.
